File: src/core/validators/CommonTypeExtension/CommonTypeExtensionIdentifierMustMatchACommonType.cs
using System.Linq;
using MetaEd.Grammar.Antlr;

namespace MetaEd.Core.Validator.CommonTypeExtension
{
    public class CommonTypeExtensionIdentifierMustMatchACommonType : ValidationRuleBase<MetaEdGrammar.CommonTypeExtensionContext>
    {
        private readonly ISymbolTable _symbolTable;

        public CommonTypeExtensionIdentifierMustMatchACommonType(ISymbolTable symbolTable)
        {
            _symbolTable = symbolTable;
        }

        public override bool IsValid(MetaEdGrammar.CommonTypeExtensionContext context)
        {
            var entityType = context.COMMON_TYPE().GetText();
            var identifier = context.extendeeName().GetText();

            return _symbolTable.IdentifiersForEntityType(entityType).Any(x => x.Equals(identifier));
        }

        public override string GetFailureMessage(MetaEdGrammar.CommonTypeExtensionContext context)
        {
            return string.Format("Common Type additions '{0}' does not match any declared Common Type.", context.extendeeName().GetText());
        }
    }
}

File: src/core/validators/CommonTypeExtension/CommonTypeExtensionIdentifierMustMatchACommonType.ts
import { ValidationRuleBase } from "../ValidationRuleBase";
import {ISymbolTable} from '../SymbolTable'
export class CommonTypeExtensionIdentifierMustMatchACommonType extends ValidationRuleBase<MetaEdGrammar.CommonTypeExtensionContext>
{
    private symbolTable: ISymbolTable;
    constructor(symbolTable: ISymbolTable) {
        super();
        this.symbolTable = symbolTable;
    }
    public isValid(context: MetaEdGrammar.CommonTypeExtensionContext): boolean {
        let entityType = context.COMMON_TYPE().getText();
        let identifier = context.extendeeName().getText();
        return this.symbolTable.identifiersForEntityType(entityType).Any(x => x.Equals(identifier));
    }
    public getFailureMessage(context: MetaEdGrammar.CommonTypeExtensionContext): string {
        return `Common Type additions '${context.extendeeName().getText()}' does not match any declared Common Type.`;
    }
}
